### DISCUSSION TITLE
Forward auth

### DISCUSSION THREAD

**Discussion by CarrySauce** [fief-dev/fief#367] [Q&A]
Some of popular proxies have forward auth functionaly, that basically forward cookie or token (headers) to auth service.

https://doc.traefik.io/traefik/middlewares/http/forwardauth/

After parsing request headers from proxy auth service will actually decide if user is authorized and allowed to access resource, or redirect user to authentication page.

This functionality is very useful for services not compatible with oauth and hidden behind proxy.
Auth service (in our case fief) can also pass headers with groups, roles, etc so they can be used on backend that is using “forwarded auth”.

I thinks it is possible to implement forward auth as external service based on fief or directly in fief.

Thoughts ?

**Comment by fief-bailiff** [comment]
Hail, @notnooblord 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to [star our repository](https://github.com/fief-dev/fief) ⭐️

<p align="center">
<img width="656" alt="star-fief" src="https://github.com/fief-dev/fief/assets/1144727/b2db7f59-102f-47e5-9751-67fa10372c52">
</p>

## Want to support us?

Subscribe to one of our paid plan to help us continue our work and receive exclusive information and benefits! Starts at $5/month 🪙

<p align="center">
    <a href="https://polar.sh/fief-dev">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/embed/subscribe.svg?org=fief-dev&darkmode=1">
      <img alt="Subscribe" src="https://polar.sh/embed/subscribe.svg?org=fief-dev">
    </picture>
    </a>
</p>

Farewell!

**Comment by frankie567** [comment]
Hello @notnooblord 👋 

That's *super* interesting, and I thought about that too. I've used [OAuth2 Proxy](https://github.com/oauth2-proxy/oauth2-proxy) in the past with Traefik forward auth pattern and found it very neat.

I think it would be nice to have this in Fief. It would require us to:

* Implement a dedicated "forwardauth" endpoint that would:
    * Check the provided token and return the authentication headers
    * *or* redirect to a login flow
* This endpoint should probably accept a `scope` as query parameters (to ask for the right scope when authenticating user)
* Add a documentation example to show how to configure Traefik and a sample backend reading those headers

I'll add this to the backlog, really love the idea :)
